libraries/opensk/src/api/persist.rs

```diff
@@ -262,17 +264,17 @@ pub trait Persist {
     /// available. This includes cases of offset being beyond the stored array.
     ///
     /// The buffer is passed in when writing is in process.
-    fn get_large_blob(
-        &self,
+    fn get_large_blob<'a>(
+        &'a self,
         mut offset: usize,
         byte_count: usize,
-        buffer: Option<&Vec<u8>>,
-    ) -> CtapResult<Option<Vec<u8>>> {
+        buffer: Option<&'a LargeBlobBuffer>,
+    ) -> CtapResult<Option<Cow<[u8]>>> {
```

Comment:
We have to be explicit otherwise it will use the lifetime of self.

```suggestion
    ) -> CtapResult<Option<Cow<'a, [u8]>>> {
```

Reply:
See above, done.